pulp_ansible/tests/functional/api/test_rbac.py

```diff
@@ -0,0 +1,248 @@
+import pytest
+import uuid
+
+from pulp_ansible.tests.functional.constants import ANSIBLE_GALAXY_URL
+
+from pulp_ansible.tests.functional.utils import gen_ansible_client, gen_ansible_remote
+
+from pulp_smash.pulp3.bindings import monitor_task
+from pulp_smash.pulp3.utils import gen_distribution, gen_repo
+
+from pulpcore.client.pulp_ansible import (
+    ApiException,
+    AsyncOperationResponse,
+    DistributionsAnsibleApi,
+    RepositoriesAnsibleApi,
+    RepositoriesAnsibleVersionsApi,
+    RemotesCollectionApi,
+    RemotesGitApi,
+    RemotesRoleApi,
+)
+
+
+def try_action(user, client, action, outcome, *args, **kwargs):
+    action_api = getattr(client, f"{action}_with_http_info")
+    try:
+        with user:
+            response, status, _ = action_api(*args, **kwargs, _return_http_data_only=False)
+        if isinstance(response, AsyncOperationResponse):
+            response = monitor_task(response.task)
+    except ApiException as e:
+        assert e.status == outcome, f"{e}"
+    else:
+        assert status == outcome, f"User performed {action} when they shouldn't been able to"
+        return response
+
+
+def test_ansible_repository_rbac(gen_user):
+    ansible_client = gen_ansible_client()
+    repo_api = RepositoriesAnsibleApi(ansible_client)
```

Comment:
Ok I figured out why the tests are failing, it's because you are not using the pytest client fixtures. You can find all the defined clients here: https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/tests/functional/conftest.py
```suggestion
def test_ansible_repository_rbac(gen_user, ansible_repo_api_client):
```